app_dart/lib/src/request_handlers/test_ownership.dart

```diff
@@ -0,0 +1,210 @@
+// Copyright 2023 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:cocoon_service/src/request_handlers/flaky_handler_utils.dart';
+import '../../protos.dart' as pb;
+
+abstract class TestOwner {
+  factory TestOwner(BuilderType builderType) {
+    switch (builderType) {
+      case BuilderType.devicelab:
+        return DeviceLabTestOwner();
+      case BuilderType.firebaselab:
+        return FirebaseLabTestOwner();
+      case BuilderType.frameworkHostOnly:
+        return FrameworkHostOnlyTestOwner();
+      case BuilderType.shard:
+        return ShardTestOwner();
+      default:
+        return UnknownTestOwner();
+    }
+  }
+
+  TestOwnership getTestOwnership(
+    pb.Target target,
+    String testOwnersContent,
+  );
+}
+
+Team teamFromString(String teamString) {
+  switch (teamString) {
+    case 'flutter/framework':
+      return Team.framework;
+    case 'flutter/engine':
+      return Team.engine;
+    case 'flutter/tool':
+      return Team.tool;
+    case 'flutter/web':
+      return Team.web;
+  }
+  return Team.unknown;
+}
+
+String getTestNameFromTargetName(String targetName) {
+  // The builder names is in the format '<platform> <test name>'.
+  final List<String> words = targetName.split(' ');
+  return words.length < 2 ? words[0] : words[1];
+}
```

Comment:
Shall we remove this function?

Reply:
I do believe that we still need this for the non devicelab tests as they still use the target name to split.